### PR TITLE
Fix go.mod and github actions to sync with supported release versions of pipeline and kubernetes

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
NONE
```

### CODE REVIEW

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -17,8 +17,8 @@ jobs:
       fail-fast: false # Keep running if one leg fails.
       matrix:
         # Keep in sync with the list of supported releases: https://kubernetes.io/releases/
+        # TODO: add 1.31.x once it is added in https://github.com/sigstore/scaffolding/releases
```
Comment (lcarva) [inline .github/workflows/kind-e2e.yaml:20]: How do I check which versions are supported?

File: .github/workflows/kind-e2e.yaml [hunk 2]
```diff
@@ -31,11 +31,11 @@ jobs:
       fail-fast: false # Keep running if one leg fails.
       matrix:
         pipelines-release:
```
Comment (lcarva) [inline .github/workflows/kind-e2e.yaml:33]: Can you add a comment that this should follow the list of versions from [here](https://github.com/tektoncd/pipeline/blob/main/releases.md#release)?

File: .github/workflows/kind-e2e.yaml [hunk 3]
```diff
@@ -31,11 +31,11 @@ jobs:
...
-        - v0.44.4  # LTS
-        - v0.47.3  # LTS
-        - v0.50.1  # LTS
-        - v0.51.0
+        - v0.50.5  # LTS
+        - v0.53.5  # LTS
+        - v0.56.3  # LTS
+        - v0.59.2  # LTS
```
Comment (lcarva) [inline .github/workflows/kind-e2e.yaml:37]: I think we should have `v0.61.0` in the list as that is the latest one.

**Review by lcarva — APPROVED** [review]

**Comment by lcarva** [comment]
Not sure what's up with all the linting failures. It might be checking the files under the `vendor` directory? Let's configure the linter to ignore those.

File: .github/workflows/kind-e2e.yaml (hunk 1 above)
Comment (PuneetPunamiya) [reply .github/workflows/kind-e2e.yaml:20]: Github action uses this [script](https://github.com/sigstore/scaffolding/blob/main/hack/setup-kind.sh#L68-L94) to install kind cluster and if fetches from the version specified here https://github.com/tektoncd/chains/blob/main/.github/workflows/reusable-e2e.yaml#L30
And here I did not see 1.31.x

File: .github/workflows/kind-e2e.yaml (hunk 2 above)
Comment (PuneetPunamiya) [reply .github/workflows/kind-e2e.yaml:33]: Yeah make sense, will add 👍🏻

File: .github/workflows/kind-e2e.yaml (hunk 3 above)
Comment (PuneetPunamiya) [reply .github/workflows/kind-e2e.yaml:37]: Yeah, but if you see the end of life, it is already gone, hence I did not add

v0.61

    Latest Release: [v0.61.0](https://github.com/tektoncd/pipeline/releases/tag/v0.61.0) (2024-06-25) ([docs]
    End of Life: 2024-07-25


Shall we still add ?

Comment (lcarva) [reply .github/workflows/kind-e2e.yaml:38]: Missed that. Let's not add it then.

File: .github/workflows/kind-e2e.yaml (hunk 1 above)
Comment (lcarva) [reply .github/workflows/kind-e2e.yaml:20]: Cool. It would be great to add a little comment here pointing to the script.

**Comment by PuneetPunamiya** [comment]
Since the number of file changes exceeds, we might have to separately bump the dependencies as the golangci lint github action is failing

**Pull request closed without merging**